packages/bot-skeleton/src/scratch/dbot.js

```diff
@@ -121,6 +121,15 @@ class DBot {
 
                 this.workspace.cached_xml = { main: main_xml };
 
+                this.workspace.addChangeListener(event => {
+                    if (event.type == Blockly.Events.BLOCK_CHANGE) {
+                        const block = this.workspace.getBlockById(event.blockId);
+                        if (block && event.element == 'collapsed') {
+                            block.contextMenu = false;
+                        }
+                    }
+                });
```

Comment:
@maryia-matskevich-deriv, can we reuse any of the existing ones instead of adding another new listener? There is one listener that is only working for `endDrag` which could be a really good candidate for this. 🙏

Reply:
yes, I've improved it, thanks